tests/libwait/wait.go

```diff
@@ -39,29 +37,92 @@ import (
 	"kubevirt.io/kubevirt/tests/watcher"
 )
 
-// WaitForVMIStartOrFailed blocks until the specified VirtualMachineInstance reached either Failed or Running states
-func WaitForVMIStartOrFailed(obj runtime.Object, seconds int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	return waitForVMIPhase(ctx, []v1.VirtualMachineInstancePhase{v1.Running, v1.Failed}, obj, seconds, wp, true)
+const defaultTimeout = 360
+
+// Option represents an action that enables an option.
+type Option func(waiting Waiting)
```

Comment:
this should be `type Option func(waiting *Waiting)` otherwise you always pass the struct by value and any change made to the object is not persisted because you create a copy of the struct every time an `Option` func is called.

Reply:
Ohh my, how did I miss this nasty bug?? :/